### PR TITLE
export migration

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan @czue some more prep work for export migrations. still working out the best way to convert RemoteApps correctly 🐟 

cc: @esoergel @orangejenny

### CODE REVIEW

File: corehq/apps/export/tests/test_export_conversion.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from django.test import TestCase, SimpleTestCase
 
 from dimagi.utils.couch.undo import DELETED_SUFFIX
+from dimagi.utils.couch.database import safe_delete
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_conversion.py]: F401 'safe_delete' imported but unused

File: corehq/apps/export/tests/test_export_conversion.py [hunk 2]
```diff
@@ -17,6 +18,7 @@
     FormExportInstance,
     CaseExportInstance,
 )
+from corehq.apps.domain.dbaccessors import get_doc_ids_in_domain_by_type
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_conversion.py]: F401 'get_doc_ids_in_domain_by_type' imported but unused

File: corehq/apps/export/models/new.py [hunk 3]
```diff
@@ -233,7 +233,7 @@ def create_default_from_export_item(table_path, item, app_ids_and_versions):
                 help_text=_(u'The ID of the associated {} case type').format(item.case_type),
                 **constructor_args
             )
-        elif feature_previews.SPLIT_MULTISELECT_CASE_EXPORT.enabled(get_request().domain):
+        elif get_request() and feature_previews.SPLIT_MULTISELECT_CASE_EXPORT.enabled(get_request().domain):
```
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: little worried about converting exports with people who use split multiselects feature preview...

File: corehq/apps/export/utils.py [hunk 4]
```diff
@@ -72,7 +80,14 @@ def convert_saved_export_to_export_instance(domain, saved_export, dryrun=False):
         if new_table:
             new_table.label = old_table.display
             new_table.selected = True
+            migration_meta.skipped_tables.append(ConversionMeta(
+                path=old_table.index,
+            ))
         else:
+            migration_meta.skipped_tables.append(ConversionMeta(
```
Comment (benrudolph) [inline corehq/apps/export/utils.py]: this is fixed in later commit to properly add to converted_tables

File: corehq/apps/export/utils.py [hunk 5]
```diff
@@ -81,72 +96,106 @@ def convert_saved_export_to_export_instance(domain, saved_export, dryrun=False):
...
+                column_path = _convert_index_to_path_nodes(index)
+                # The old style column indexes always look like they contains no repeats,
+                # so replace that parts that could be repeats with the table path
+                column_path = table_path + column_path[len(table_path):]
+
+                system_property = _get_system_property(
+                    column.index,
+                    transform,
+                    export_type,
+                    new_table.path
```
Comment (benrudolph) [inline corehq/apps/export/utils.py]: w=1 is helpful for this commit: https://github.com/dimagi/commcare-hq/pull/12649/commits/d6c2ae84531f8b8dc652281992ac103a1e59aa59?w=1

File: corehq/apps/export/utils.py [hunk 6]
```diff
@@ -72,7 +80,14 @@ def convert_saved_export_to_export_instance(domain, saved_export, dryrun=False):
         if new_table:
             new_table.label = old_table.display
             new_table.selected = True
+            migration_meta.skipped_tables.append(ConversionMeta(
```
Comment (benrudolph) [inline corehq/apps/export/utils.py]: this is fixed to be in converted_tables

File: corehq/apps/export/models/new.py [hunk 7]
```diff
@@ -1610,6 +1610,24 @@ def get_value(self, domain, doc_id, doc, base_path, **kwargs):
...
         return values
 
+
+class ConversionMeta(DocumentSchema):
+    path = StringProperty()
+    failure_reason = StringProperty()
+    info = ListProperty()
+
+
+class ExportMigrationMeta(Document):
```
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: 👍 good call. gonna do a couple dry runs on staging to see what other useful information would be good to add

**Pull request merged**